tree_in_context/src/utility/in_context.egg

```diff
@@ -2,54 +2,41 @@
 ; It gives each program path a unique equality relation.
 ; This can be quite expensive, so be careful running these rules.
 
-(ruleset in_context)
+(ruleset context_propagate)
+(ruleset context_creation)
 
 (sort InContextList)
 
-; In order to saturate and not create unecessary contexts, we need to collapse
-; duplicate assumptions.
-; For example, the rewrite over `Function` could create many
-; `(InContext (InFunc name) (InContext (InFunc name) ...))` nestings
-(rewrite (InContext assumption (InContext assumption rest))
+; Here's a helpful rule that chooses a more specific context over
+; an inner, less specific one.
+(rewrite (InContext assumption (InContext assumption2 rest))
```

Comment:
Will this rule still fire if we ban creating new InContext when subfield have InContext?

Reply:
It may still fire if other rules create contexts